python/py_rust_library.bzl

```diff
@@ -0,0 +1,51 @@
+"""
+"""
+
+load("@bazel_skylib//rules:copy_file.bzl", "copy_file")
+load("@rules_python//python:python.bzl", "py_library")
+load("@rules_rust//rust:rust.bzl", "rust_library")
+
+def py_rust_library(name, **kwargs):
+    """
+    Wraps a rust shared library that implements a python native extension to be usable by @rules_python targets.
+
+    Typically this is done through the use of a crate such as `pyo3` or `cpython`.
+
+    Args:
+        name (str): A unique name for this target.
+        **kwargs:   Passed directly to rust_library.
+    """
+
+    mangled = name + "_mangled_so"
+    rust_library(
+        name = mangled,
+        crate_type = "cdylib",
```

Comment:
One problem with this macro is that rules_rust doesn't have the equivalent of `linkstatic = False` as in the (naieve?) cc macro https://github.com/bazelbuild/bazel/issues/701#issuecomment-849157478

Reply:
https://github.com/mfarrugi/rules_rust/commit/a483a7f096ff57f54534197b94b6b4ee7d74c09e illustrates the issue, and fails like so:
```
Traceback (most recent call last):
  File "/home/marco/.cache/bazel/_bazel_marco/9ec96d05a32431fe9b2731f854ff9adb/execroot/examples/bazel-out/k8-fastbuild/bin/ffi/python_calling_rust/test.runfiles/examples/ffi/python_calling_rust/test.py", line 12, in <module>
    assert s1 == s2, (s1, s2)
AssertionError: (-877036644, -885769300)
```

The same thing can happen with the cc macro linked above if there is a dependency that can only be linked statically. Fixing this pedantically would require each py_binary rule to link all of its native extensions at the same time (or require them to be fully dynamically linked), which is far beyond the scope of this PR.